sse2neon.h

```diff
@@ -8358,6 +8359,495 @@ FORCE_INLINE int _mm_testz_si128(__m128i a, __m128i b)
+                vec1, vec_minusone, vreinterpretq_u##size##_m128i(mtx[j])));   \
+        }                                                                      \
+        for (int j = lb; j < bound; j++) {                                     \
+            mtx[j] = vreinterpretq_m128i_u##size(                              \
+                vbslq_u##size(vec1, vec_minusone, vec_zero));                  \
+        }                                                                      \
+        unsigned SSE2NEON_IIF(data_type)(char, short) *ptr =                   \
+            (unsigned SSE2NEON_IIF(data_type)(char, short) *) mtx;             \
+        for (int i = 0; i < bound; i++) {                                      \
+            int val = 1;                                                       \
+            for (int j = 0, k = i; j < bound - i && k < bound; j++, k++)       \
+                val &= ptr[k * bound + j];                                     \
+            res += val << i;                                                   \
+        }                                                                      \
+        return res;                                                            \
+    }
+
+SSE2NEON_AGGREGATE_EQUAL_ORDER_IMPL(8, 16, UBYTE)
+SSE2NEON_AGGREGATE_EQUAL_ORDER_IMPL(16, 8, UWORD)
+
+#undef UBYTE
+#undef UWORD
+
+SSE2NEON_CMP_EQUAL_ORDERED_IMPL(byte, 8, 16)
+SSE2NEON_CMP_EQUAL_ORDERED_IMPL(word, 16, 8)
+
+#define SSE2NEON_CMPESTR_LIST                              \
+    _(CMP_UBYTE_EQUAL_ANY, 0, cmp_byte_equal_any)          \
+    _(CMP_UWORD_EQUAL_ANY, 1, cmp_word_equal_any)          \
+    _(CMP_SBYTE_EQUAL_ANY, 2, cmp_byte_equal_any)          \
+    _(CMP_SWORD_EQUAL_ANY, 3, cmp_word_equal_any)          \
+    _(CMP_UBYTE_RANGES, 4, cmp_ubyte_ranges)               \
+    _(CMP_UWORD_RANGES, 5, cmp_uword_ranges)               \
+    _(CMP_SBYTE_RANGES, 6, cmp_sbyte_ranges)               \
+    _(CMP_SWORD_RANGES, 7, cmp_sword_ranges)               \
+    _(CMP_UBYTE_EQUAL_EACH, 8, cmp_byte_equal_each)        \
+    _(CMP_UWORD_EQUAL_EACH, 9, cmp_word_equal_each)        \
+    _(CMP_SBYTE_EQUAL_EACH, 10, cmp_byte_equal_each)       \
+    _(CMP_SWORD_EQUAL_EACH, 11, cmp_word_equal_each)       \
+    _(CMP_UBYTE_EQUAL_ORDERED, 12, cmp_byte_equal_ordered) \
+    _(CMP_UWORD_EQUAL_ORDERED, 13, cmp_word_equal_ordered) \
+    _(CMP_SBYTE_EQUAL_ORDERED, 14, cmp_byte_equal_ordered) \
+    _(CMP_SWORD_EQUAL_ORDERED, 15, cmp_word_equal_ordered)
+
+enum {
+#define _(name, number, func_suffix) name = number,
+    SSE2NEON_CMPESTR_LIST
+#undef _
+};
+typedef int (*cmpestr_func_t)(__m128i a, int la, __m128i b, int lb);
+static cmpestr_func_t _sse2neon_cmpfunc_table[] = {
+#define _(name, number, func_suffix) [name] = _sse2neon_##func_suffix,
+    SSE2NEON_CMPESTR_LIST
+#undef _
+};
+
+static inline int neg_fun(int res, int lb, int imm8, int bound)
```

Comment:
Let's rename this function to avoid potential conflicts. How about `_sse2neon_sido_negative`?